react-components/src/components/RevealToolbar/LayersContainer/Image360LayersContainer.tsx

```diff
@@ -74,19 +77,24 @@ export const Image360CollectionLayerContainer = (): ReactElement => {
   };
 
   return (
-    <Menu.Submenu content={image360Content()} title="360 images">
-      <Flex direction="row" justifyContent="space-between">
-        <Checkbox
-          checked={all360ImagesVisible}
-          indeterminate={indeterminate}
-          onChange={(e, c) => {
-            e.stopPropagation();
-            handleAll360ImagesVisibility(c as boolean);
-          }}
-        />
-        <StyledLabel> 360 images </StyledLabel>
-        <StyledChipCount label={count} hideTooltip type="neutral" />
-      </Flex>
-    </Menu.Submenu>
+    <>
+      {image360LayerData.length > 0 && (
+        <Menu.Submenu content={image360Content()} title="360 images">
+          <Flex direction="row" justifyContent="space-between">
+            <Checkbox
+              key={`all360ImagesCheckbox-${String(someImagesVisible)}-${String(indeterminate)}`}
```

Comment:
Do we need a `key` property here? Isn't that only necessary when we have multiple similar elements inside the same parent?

Reply:
Checkbox needs to trigger re-render when either `someImagesVisible` or `indeterminate` value changes, so when `key` changes, it would trigger the re-render on the component